### PR TITLE
List user repositories api

### LINKED ISSUE #194: Add support for the list user repositories endpoint
Can't find api to do this.

The `Github().repos.listOrgRepos` can just get repos of an organization.

### PR DESCRIPTION
**Pull request by bhop**
fixes #194

### CODE REVIEW

File: github4s/shared/src/main/scala/github4s/api/PullRequests.scala [hunk 1]
```diff
@@ -28,62 +28,59 @@ import scala.language.higherKinds
 
 /** Factory to encapsulate calls related to PullRequests operations  */
 class PullRequests[C, M[_]](
-  implicit urls: GithubApiUrls,
-  C: Capture[M],
-  httpClientImpl: HttpRequestBuilderExtension[C, M]) {
+    implicit urls: GithubApiUrls,
+    C: Capture[M],
+    httpClientImpl: HttpRequestBuilderExtension[C, M]) {
```
Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/PullRequests.scala]: could you turn off the auto-indent here (or change it) as we'd like to keep the same indentation across the project?

File: github4s/shared/src/main/scala/github4s/api/Repos.scala [hunk 2]
```diff
@@ -75,6 +75,31 @@ class Repos[C, M[_]](
...
+   * @param `type` visibility of the retrieved repositories, can be "all", "public", "private",
+   * "forks", "sources" or "member"
+   * @return GHResponse[List[Repository]] the list of repositories for this user
+   */
+  def listUserRepos(
+      accessToken: Option[String] = None,
+      headers: Map[String, String] = Map(),
+      user: String,
+      `type`: Option[String] = None,
+      pagination: Option[Pagination] = Some(httpClient.defaultPagination)): M[
```
Comment (BenFradet) [inline github4s/shared/src/main/scala/github4s/api/Repos.scala]: could you do:

```scala
): M[GHResponse[List[Repository]]]
```

as above?

File: github4s/shared/src/main/scala/github4s/api/PullRequests.scala (hunk 1 above)
Comment (BenFradet) [reply github4s/shared/src/main/scala/github4s/api/PullRequests.scala]: this is true for a lot of the files below too

**Comment by juanpedromoreno** [comment]
@bhop Yup, the integrations tests failed because only the maintainers have access to the API key. In those cases, we need to run them manually.

File: github4s/shared/src/main/scala/github4s/api/Repos.scala (hunk 2 above)
Comment (bhop) [reply github4s/shared/src/main/scala/github4s/api/Repos.scala]: Done

File: github4s/shared/src/main/scala/github4s/api/PullRequests.scala (hunk 1 above)
Comment (bhop) [reply github4s/shared/src/main/scala/github4s/api/PullRequests.scala]: Fixed

**Comment by BenFradet** [comment]
Cool, shout when you need a release :+1:

**Pull request merged**